aom/src/main/java/com/nedap/archie/aom/CPrimitiveObject.java

```diff
@@ -85,6 +85,9 @@ public boolean isValidValue(ValueType value) {
      */
     public boolean isValidValue(ModelInfoLookup lookup, Object value) {
         Object convertedValue = lookup.convertToConstraintObject(value, this);
+        if(convertedValue == null) {
```

Comment:
Are you sure this change does not cause problems?
I tried to check, and this should not change our normal validation. It may affect `matches {...}` rule evaluation though, and could affect tuple validation?

Reply:
Good point, I'm not sure. We should check this.

Reply:
I found another solution by returning an empty `TerminologyCode` object.